### PR TITLE
Carousel API doc

### PR DESCRIPTION
**Pull request by alavers**
@mspensieri @Mario54 @lemieux @jpjoyal @liyefei737 @chloepouprom

### CODE REVIEW

File: source/rest/index.md [hunk 1]
```diff
@@ -1103,29 +1103,30 @@ smooch.conversations.sendMessage('c7f6e6d6c3a637261bd9656f', {
...
-| **actions**<br/>*optional* | An array of action buttons (see section below) |
-| **payload**<br/>*optional* | The payload of a `reply` action, if applicable |
-
-### Action buttons
-Actions buttons can be sent through the API by including them in the message payload.
+| **Arguments**                |                       |
+|------------------------------|-----------------------|
+| **role**<br/><span class='req'>required</span>       | The role of the individual posting the message. Can be either `appUser` or `appMaker`. |
+| **text**<br/><span class='mut'>optional*</span>      | The text content of the message. `text` becomes optional if either `mediaUrl` or `actions` are provided. Mututally exclusive with `items`. |
+| **actions**<br/><span class='mut'>optional*</span>   | Array of [action buttons](#action-buttons). Mututally exclusive with `items`. |
```
Comment (dannytranlx) [inline source/rest/index.md]: Mutually*?

File: source/rest/index.md [hunk 2]
```diff
@@ -1270,15 +1271,189 @@ smooch.conversations.sendMessage('c7f6e6d6c3a637261bd9656f', {
...
+
+| **Arguments**                |                        |
+|------------------------------|------------------------|
+| **title**<br/><span class='req'>required</span>       | The title of the carousel item. |
+| **actions**<br/><span class='req'>required</span>     | Array of carousel actions. At least 1 is required, a maximum of 3 are allowed. Only `link` and `postback` actions are supported. |
+| **description**<br/><span class='opt'>optional</span> | The text description, or subtitle. |
+| **mediaUrl**<br/><span class='opt'>optional</span>   | The image URL to be shown in the carousel item. |
+| **mediaType**<br/><span class='opt'>optional</span>   | If a `mediaUrl` was specified, the media type is defined here, for example `image/jpeg`. If `mediaType` is not specified, the media type will be resolved with the `mediaUrl`. |
+
+### Carouel Actions
```
Comment (dannytranlx) [inline source/rest/index.md]: Carousel*

File: source/rest/index.md [hunk 3]
```diff
@@ -1270,15 +1271,189 @@ smooch.conversations.sendMessage('c7f6e6d6c3a637261bd9656f', {
...
+
+Carousel actions have the same structure as [action buttons](#action-buttons), however only `link` and `postback` action types are permitted.
+
+### Channel Suport
```
Comment (dannytranlx) [inline source/rest/index.md]: Support*

File: source/rest/index.md [hunk 4]
```diff
@@ -1270,15 +1271,189 @@ smooch.conversations.sendMessage('c7f6e6d6c3a637261bd9656f', {
...
+1. Tacos
+Description
+More info http://example.org
+
+2. Ramen
+Description
+More info http://example.org
+```
+
+Carousel messges are a horizontally scrollable set of items that may each contain text, an image, and action buttons. Not all messaging channels fully support carousel messages; currently only Facebook Messenger and Telegram cover the full functionality. For all other platforms a carousel message is rendered as raw text. The raw text fallback does not include any images or postback action buttons.
```
Comment (jpjoyal) [inline source/rest/index.md]: messges -> messages

**Comment by dannytranlx** [comment]
👍

File: source/rest/index.md [hunk 5]
```diff
@@ -1270,15 +1271,189 @@ smooch.conversations.sendMessage('c7f6e6d6c3a637261bd9656f', {
...
+
+<aside class="notice">
+Only the `appMaker` role can used to send a carousel message.
```
Comment (jpjoyal) [inline source/rest/index.md]: be used

**Comment by jpjoyal** [comment]
👍

**Pull request merged**